pkg/operator/cluster/ceph_user.go

```diff
@@ -0,0 +1,68 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/ceph/client"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newCephUser(context *clusterd.Context, ns string) *cephUser {
+	n := fmt.Sprintf("%s-rook-user", ns)
```

Comment:
The design is to have one ceph user per namespace (decoupling for security). So each namespace that uses a rook resource would get a secret with specific ceph token.
The secret name is the same in each namespace, the token and username for ceph will be different. The controller validates if a secret for a particular cluster exists in the clientNamespace, and creates it if it doesn't.
Basically, following your convention `targetNamespace-rook-user: ClientNamespaceToken` in each namespace that uses rook resources. Appending `clientNamespace` to the name of the secret in a clientNamespace seemed redundant.